### PR TITLE
Fix Taskable assigned_attorney method to ignore Docket Switch attorneys

### PR DESCRIPTION
**Pull request by nanotone**
### Description
This PR fixes a bug in the implementation of `Taskable#assigned_attorney` so that it never reports an attorney solely assigned to a Docket Switch task.

The `assigned_attorney` denotes the attorney tasked with drafting an appeal's decision, and is saved on the associated JudgeCaseReview record to give that attorney "credit" for their work. Therefore it is important to ensure that the correct attorney is being returned.

[Slack thread](https://dsva.slack.com/archives/C54QCEHAL/p1631806550182100) where the investigation into the bug happened.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Added a new test for `Appeal#assigned_attorney`
2. New test fails on old application logic

### CODE REVIEW

File: app/models/concerns/taskable.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ def assigned_attorney
     tasks.not_cancelled
       .order(created_at: :desc)
       .includes(:assigned_to)
-      .detect { |t| t.is_a?(AttorneyTask) }
+      .detect { |t| t.is_a?(AttorneyTask) && !t.is_a?(DocketSwitchAbstractAttorneyTask) }
```
Comment (eileen-nava) [inline app/models/concerns/taskable.rb:10]: 👍

**Review by eileen-nava — APPROVED** [review]
LGTM!

File: spec/models/appeal_spec.rb [hunk 2]
```diff
@@ -980,6 +981,19 @@
         task2.cancelled!
         expect(subject).to eq attorney
       end
+
+      context "when there is a more recent DocketSwitch attorney task" do
+        let(:judge) { create(:user, :with_vacols_judge_record, full_name: "Judge the First", css_id: "JUDGE_1") }
+        let(:root_task) { create(:root_task, appeal: appeal) }
+        let!(:ds_task) do
+          create(:docket_switch_denied_task, parent: root_task, appeal: appeal, assigned_to: attorney,
+                                             assigned_by: judge, created_at: 1.minute.ago)
```
Comment (karinamzalez) [inline spec/models/appeal_spec.rb:990]: would it be worth noting via a code comment here that this creation date is intentionally after that of task2 or is enough context implied with the test title?

**Review by karinamzalez — APPROVED** [review]
👍🏽 -- just a minor Q from me, but looks good :)

File: spec/models/appeal_spec.rb (hunk 2 above)
Comment (nanotone) [reply spec/models/appeal_spec.rb:990]: I think the test title should be enough context, and in general RSpec contexts should be informative enough to not require additional comments. Lemme know if you think of any improvements to the title though!

**Pull request merged**